cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/dataset/MetadataDatasetTest.java

```diff
@@ -1137,42 +1137,34 @@ public void apply() throws Exception {
         searchResults = dataset.search(namespaceId, "*", targets, nameAsc, 0, 2, 0, null, false,
                                        EnumSet.allOf(EntityScope.class));
         Assert.assertEquals(ImmutableList.of(flowEntry, dsEntry), searchResults.getResults());
-        Assert.assertEquals(ImmutableList.of(flowEntry, dsEntry, appEntry), searchResults.getAllResults());
```

Comment:
you mean if offset = 2, limit = 2, numCursors = 2
results will have (limit + 1) * numCursors = 6 elements, but all results will have offset + results = 8 elements? 

To count the total, you don't actually need to return the results, right? You can just add the offset?